examples/sign-leaf-with-ca.rs

```diff
@@ -0,0 +1,99 @@
+use rcgen::{
+	date_time_ymd, BasicConstraints, Certificate, CertificateParams, CertificateSigningRequest,
+	DnType, ExtendedKeyUsagePurpose, IsCa, KeyUsagePurpose,
+};
+use time::{Duration, OffsetDateTime};
+use x509_parser::certification_request::X509CertificationRequest;
+use x509_parser::prelude::FromDer;
+
+/// Example demonstrating signing end-endity certificate with ca
+fn main() {
+	let ca = Ca::new();
+	let entity = Entity::new();
+
+	println!("directly signed end-entity certificate:");
+	let direct = entity
+		.certificate
+		.serialize_pem_with_signer(&ca.certificate)
+		.unwrap();
+	println!("{direct}");
+
+	println!("ca certificate:");
+	let pem = ca.certificate.serialize_pem().unwrap();
+
+	println!("{}", pem);
+}
+
+struct Ca {
```

Comment:
I think these types aren't adding very much and `Ca::new` and `Entity::new` could be written more clearly as free-standing `new_ca` and `new_end_entity` fn's. WDYT?

Reply:
I agree